vaadin-notification-flow-parent/vaadin-notification-flow/src/main/java/com/vaadin/flow/component/notification/Notification.java

```diff
@@ -572,6 +580,10 @@ private void attachComponentTemplate() {
     protected void onAttach(AttachEvent attachEvent) {
         super.onAttach(attachEvent);
         initConnector();
+        if (refreshed) {
+            open();
+        }
+        refreshed = true;
```

Comment:
I wonder if this new state is really necessary, instead of using the opened state? I think this would re-open any notification that has been manually added to the UI, even if it has already closed, which doesn't seem right.

I also think we might want to move this into a separate change. Fixing the exception is an obvious thing to do, but reopening the notification on preserve on refresh is more opinionated and could have side effects.